src/main/scala/advent/solutions/Day3.scala

```diff
@@ -64,7 +233,26 @@ object Day3 {
       *         or [[None]] if they don't intersect at all
       */
     def distanceToIntersection(wire0: Wire, wire1: Wire): Option[Int] = {
-      ???
+      val wire0Coordinates: List[Coordinate] =
+        pairEachDisplacementWithStartingAndEndingCoordinates(
+          wire0
+        ).flatMap(i =>
+          pathTravelled(i.displacement, i.startingAndEndingPoints).points
+        )
+
+      val wire1Coordinates =
+        pairEachDisplacementWithStartingAndEndingCoordinates(
+          wire1
+        ).flatMap(i =>
+          pathTravelled(i.displacement, i.startingAndEndingPoints).points
+        )
+
+      val intersectionPoints = wire0Coordinates
+        .intersect(wire1Coordinates)
+        .filterNot(_ == Coordinate(0, 0))
```

Comment:
The wires might cross at (0,0) later on, but this would filter out any intersections that happened later at `(0,0)`.  It might be better if the wires didn't include their starting point.